### PR TITLE
Implement NavigateTo for single files, instead of the project.

### PR DESCRIPTION
**Pull request by paulvanbrenk**
This implements a new API which allows you to limit the NavigateTo results to a single file. This is in response to recent VS changes.

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -1344,10 +1344,13 @@ namespace ts {
         }
 
         /// NavigateTo
-        function getNavigateToItems(searchValue: string, maxResultCount?: number): NavigateToItem[] {
+        function getNavigateToItems(searchValue: string, maxResultCount?: number, fileName?: string): NavigateToItem[] {
             synchronizeHostData();
+
+            const sourceFiles = !fileName ? program.getSourceFiles() : [program.getSourceFile(fileName)];
```
Comment (sandersn) [inline src/services/services.ts]: can you invert the if so the condition is just `fileName`?

**Comment by sandersn** [comment]
:+1:

File: src/services/services.ts (hunk 1 above)
Comment (paulvanbrenk) [inline src/services/services.ts]: Sure

**Comment by vladima** [comment]
:+1:

**Comment by mhegazy** [comment]
👍

**Comment by mhegazy** [comment]
We need to port this to release-2.0.5 branch at some point in the future. and also make the corresponding VS changes to call the new API.

**Pull request merged**